File: 백준/Gold/1377. 버블 소트/버블 소트.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
//        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(br.readLine());
        Node[] nums = new Node[N];
        for(int i=0; i<N; i++) {
            int t = Integer.parseInt(br.readLine());
            nums[i] = new Node(t, i);
        }

        Arrays.sort(nums);

        int ans = 0;
        for(int i=0; i<N; i++) {
            ans = Math.max(ans, nums[i].idx-i);
        }
        
        ans++;

        sb.append(ans);

        System.out.println(sb);
        br.close();
    }

    static class Node implements Comparable<Node>{
        int value, idx;
        public Node(int value, int idx) {
            this.value = value;
            this.idx = idx;
        }

        @Override
        public int compareTo(Node o1) {
            return this.value - o1.value;
        }
    }
}

File: 백준/Gold/1377. 버블 소트/버블 소트.cc
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

struct Node {
    int value;
    int idx;
};

bool cmp(const Node&a, const Node&b) {
    if(a.value != b.value) return a.value < b.value;
    return a.idx < b.idx;
}

int main() {

    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;

    vector<Node> nums(N);

    for(int i=0; i<N; i++) {
        int temp;
        cin >> temp;
        nums[i] = {temp, i};
    }

    sort(nums.begin(), nums.end(), cmp);

    int ans = 0;
    
    for(int i=0; i<N; i++) {
        int temp = nums[i].idx - i;
        if(ans < temp) ans = temp;
    }

    cout << ans+1 << endl;
    

    return 0;
}